src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/New/CSharpFormattingPass.CSharpDocumentGenerator.cs

```diff
@@ -217,7 +220,75 @@ protected override LineInfo DefaultVisit(RazorSyntaxNode node)
             public override LineInfo VisitCSharpExpressionLiteral(CSharpExpressionLiteralSyntax node)
             {
                 Debug.Assert(node.LiteralTokens.Count > 0);
-                return VisitCSharpLiteral(node, node.LiteralTokens[^1]);
+                var lastToken = node.LiteralTokens[^1];
+
+                // Literals that span multiple lines are interesting. eg, given:
+                //
+                // <div class="@(foo
+                //              .bar)" />
+                if (_sourceText.GetLinePositionSpan(node.Span).SpansMultipleLines())
+                {
+                    var skipPreviousLine = false;
```

Comment:
Should most of the body of the block be extracted to something named like VisitMultiLineCsharpLiteral, then this particular method is just a quick check and return the result of the corresponding "VisitXCsharpLiteral"?

Reply:
Sounds reasonable